pengine/graph.c

```diff
@@ -441,17 +441,16 @@ update_colo_start_chain(action_t *action)
 
     if (is_not_set(action->flags, pe_action_runnable) && safe_str_eq(action->task, RSC_START)) {
         rsc = uber_parent(action->rsc);
+        if (rsc->parent) {
```

Comment:
This change makes no sense.
What is it based on?  Observed or theoretical problems?
At the very least the comment - which talks about stops - doesn't match the check for a start action.

Reply:
It's your comment :)

By "uber_parent() stops before the bundle" you meant that uber_parent() doesn't go up the rsc->parent chain all the way to the bundle resource. So you added this block (in the later position) to get the bundle resource when rsc is in a bundle. The problem is that where it was, it was apply even when uber_parent() wasn't called. AFAICT the only purpose of this is to adjust the uber_parent() behavior, so it should only be called when that is called.